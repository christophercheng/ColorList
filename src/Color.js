import React, {Component} from "react"
import StarRating from "./StarRating"
import PropTypes from "prop-types"
 
/*

// wondering if it's possible to replace props with just the actual prop and not the property name
const Color = ({number_stars=5, rating=0, title, color,onRemove,onRate}) => {
 	
	const colorStyle={
		backgroundColor:color,
		height:'50px',
	}
	return (
	<section className="color">

		<h1>{title}</h1>
		<button onClick={onRemove}>X</button>
		<div className="color-bar" style={ colorStyle} ></div>
		<StarRating number_stars={number_stars} rating={rating} onClickFN={onRate}/>
	</section>
	)
}

 
*/

class Color extends Component {
	static propTypes = {
		number_stars:PropTypes.number,
		title: PropTypes.string,
		rating: PropTypes.number,
		color: PropTypes.string,
		onRate: PropTypes.func,
		onRemove:PropTypes.func
	}
	componentWillMount(){
	
		this.colorStyle = {
				backgroundColor:"#CCC",
				height:'150px',
				width:'300px',
				border: "1px solid black",
				margin:'25px'
			}
		
	}

	shouldComponentUpdate(nextProps){
		const {rating}=this.props
		return rating != nextProps.rating	
	}

	render (){
	const {number_stars=5,rating=0,color,title,onRemove,onRate} = this.props
		console.log("color:", color)
		return (
		<section className="color" style={this.colorStyle}>

			<h1 ref="title">{title}<button onClick={onRemove}>X</button></h1>

			<div className="color-bar" style={{backgroundColor:color,height:50}}></div>
						<StarRating number_stars={number_stars} rating={rating} color={color} onClickFN={onRate}/>
		</section>
		)
	}

	componentWillUpdate(nextProps){
		const {title,rating} = this.props
		this.refs.title.style.backgroundColor="red"
		this.refs.title.style.color="white"
		//console.log(this.colorStyle)
		//alert(`${title}:rating ${rating} -> ${nextProps.rating}`)
		this.colorStyle={...this.colorStyle,backgroundColor:"#FFF"}
		//.log(this.colorStyle)
	}

	componentDidUpdate(prevProps){
		const {title,rating} = this.props
		const status = (rating > prevProps.rating) ? 'better' : 'worse'
		console.log(`${title} is getting ${status}`)
		this.colorStyle={...this.colorStyle,backgroundColor:"#CCC"}
	}
}

export default Color