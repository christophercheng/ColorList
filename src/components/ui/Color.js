import FaTrash from 'react-icons/lib/fa/trash-o';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import StarRating from './StarRating';
import './../../stylesheets/Color.css';

class Color extends Component {
  static defaultProps = {
    numberStars: 5,
    title: 'A color',
    rating: 0,
    color: 'Blue',
    onRate: f => f,
    onRemove: f => f,
  }

  static propTypes = {
    numberStars: PropTypes.number,
    title: PropTypes.string,
    rating: PropTypes.number,
    color: PropTypes.string,
    onRate: PropTypes.func,
    onRemove: PropTypes.func,
  }

  static contextTypes = { store: PropTypes.object }

  render() {
    const { store } = this.context;
    console.log('color store:', store);
    const {
      numberStars, rating, color, title, onRemove, onRate,
    } = this.props;
    return (
      <section className="color" style={this.style}>
        <h1 ref={(el) => { this.title = el; }}>{title}</h1>
        <button onClick={onRemove}><FaTrash /></button>
        <div className="color" style={{ backgroundColor: color }} />
        <div>
          <StarRating numberStars={numberStars} rating={rating} color={color} onClickFN={onRate} />
        </div>
      </section>
    );
  }
}


export default Color;
