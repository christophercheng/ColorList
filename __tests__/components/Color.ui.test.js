import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Color from '../../src/components/ui/Color';

describe('<Color /> UI Component Test Suite', () => {
  it('Renders correct properties...', () => {
    const wrapperOutput = shallow(
      <Color
        title="Test Color"
        color="#F0F0F1"
        rating={3}
      />,
    );
    expect(toJSON(wrapperOutput)).toMatchSnapshot();
  });
});
