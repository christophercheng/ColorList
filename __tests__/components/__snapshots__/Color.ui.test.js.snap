// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI Component Test Suite Renders correct properties... 1`] = `
<section
  className="color"
>
  <h1>
    Test Color
  </h1>
  <button
    onClick={[Function]}
  >
    <FaTrashO />
  </button>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#F0F0F1",
      }
    }
  />
  <div>
    <StarRating
      color="#F0F0F1"
      numberStars={5}
      onClickFN={[Function]}
      rating={3}
    />
  </div>
</section>
`;
